misc/python/materialize/cut-release.py

```diff
@@ -0,0 +1,66 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+"""Cut a new release and push it to the Materialize repo"""
+
+import argparse
+import sys
+
+from semver.version import Version
+
+from materialize import spawn
+from materialize.git import checkout, tag_annotated, get_branch_name, push
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog="cut_release",
+        description="Creates a new release for Materialize.",
+    )
+    parser.add_argument(
+        "--sha",
+        help="Chosen SHA of the release",
+        type=str,
+    )
+    parser.add_argument(
+        "--version",
+        help="Version of release",
+        type=str,
```

Comment:
Should be able to have argparse to the type parsing!

```suggestion
        type=Version.parse,
```